Add `consistent-existence-index-check` rule (#2425)
sindresorhus/eslint-plugin-unicorn: docs/rules/consistent-non-exists-index-check.md

> # Enforce consistent styling when checking for element existence using `indexOf()`

Reviewer: ```suggestion
# Enforce consistent style for element existence checks with `indexOf()`
```
